client/components/swipeable/index.js

```diff
@@ -0,0 +1,249 @@
+		const position = getDragPositionAndTime( event );
+		setDragStartData( position );
+		setPagesStyle( { ...pagesStyle, transitionDuration: `0ms` } ); // Set transition Duration to 0 for smooth dragging.
+	};
+
+	const hasSwipedToNextPage = ( delta ) => ( isRtl ? delta > 0 : delta < 0 );
+	const hasSwipedToPreviousPage = ( delta ) => ( isRtl ? delta < 0 : delta > 0 );
+
+	const handleDragEnd = ( event ) => {
+		if ( ! dragStartData ) {
+			return; // End early if we are not dragging any more.
+		}
+
+		const OFFSET_THRESHOLD = 100; // Number of pixels to travel before we trigger the slider to move to the desired slide.
+		const VELOCITY_THRESHOLD = 0.5;
+		const dragPosition = getDragPositionAndTime( event );
+		const delta = dragPosition.x - dragStartData.x;
+		const absoluteDelta = Math.abs( delta );
+		const velocity = absoluteDelta / ( dragPosition.timeStamp - dragStartData.timeStamp );
+
+		const hasMetThreshold = absoluteDelta > OFFSET_THRESHOLD || velocity > VELOCITY_THRESHOLD;
+
+		let newIndex = currentPage;
+		if ( hasSwipedToNextPage( delta ) && hasMetThreshold && numPages !== currentPage + 1 ) {
+			newIndex = currentPage + 1;
+		}
+
+		if ( hasSwipedToPreviousPage( delta ) && hasMetThreshold && currentPage !== 0 ) {
+			newIndex = currentPage - 1;
+		}
+		const offset = getOffset( newIndex );
+		setPagesStyle( {
+			transform: `translate3d(${ offset }px, 0px, 0px)`,
+			transitionDuration: `300ms`,
+		} );
+		onPageSelect( newIndex );
+		setDragStartData( null );
+	};
+
+	const handleDrag = ( event ) => {
+		if ( ! dragStartData ) {
+			return;
+		}
+		const dragPosition = getDragPositionAndTime( event );
+		const delta = dragPosition.x - dragStartData.x;
+
+		const offset = getOffset( currentPage ) + delta;
+
+		// Allow for swipe left or right
+		if (
+			( numPages !== currentPage + 1 && hasSwipedToNextPage( delta ) ) ||
+			( currentPage !== 0 && hasSwipedToPreviousPage( delta ) )
+		) {
+			setPagesStyle( {
+				...pagesStyle,
+				transform: `translate3d(${ offset }px, 0px, 0px)`,
+				transitionDuration: `0ms`,
+			} );
+		}
+
+		if ( ! swipeableArea ) {
+			return;
+		}
+
+		// Did the user swipe out of the swipable area?
+		if (
+			dragPosition.x < swipeableArea.left ||
+			dragPosition.x > swipeableArea.right ||
+			dragPosition.y > swipeableArea.bottom ||
+			dragPosition.y < swipeableArea.top
+		) {
+			handleDragEnd( event );
+		}
+	};
+
+	const getTouchEvents = () => {
+		if ( 'onpointerup' in document ) {
+			return {
+				onPointerDown: handleDragStart,
+				onPointerMove: handleDrag,
+				onPointerUp: handleDragEnd,
+			};
+		}
+
+		if ( 'ondragend' in document ) {
+			return {
+				onDragStart: handleDragStart,
+				onDrag: handleDrag,
+				onDragEnd: handleDragEnd,
+			};
+		}
+
+		if ( 'ontouchend' in document ) {
+			return {
+				onTouchStart: handleDragStart,
+				onTouchMove: handleDrag,
+				onTouchEnd: handleDragEnd,
+			};
+		}
+
+		return null;
+	};
+
+	return (
+		<div>
```

Comment:
Do we need that `div`? Could it be a `<>...</>`?

Reply:
Good call.